Vagrantfile

```diff
@@ -61,49 +65,91 @@ Vagrant.configure(2) do |config|
     export CHE_LOCAL_CONF_DIR=/home/vagrant/.che &>/dev/null
     cp /home/vagrant/eclipse-che-*/conf/che.properties /home/vagrant/.che/che.properties
-    sed -i "s/\${catalina.base}\/temp\/local-storage/\/home\/vagrant\/.che/g" /home/vagrant/.che/che.properties
-    sed -i "s|\${che.home}/workspaces|/home/vagrant/.che|" /home/vagrant/.che/che.properties
+    sed -i 's|${catalina.base}/temp/local-storage|/home/vagrant/.che|' /home/vagrant/.che/che.properties
+    sed -i 's|${che.home}/workspaces|/home/vagrant/.che|' /home/vagrant/.che/che.properties
     echo 'export CHE_LOCAL_CONF_DIR=/home/vagrant/.che' >> /home/vagrant/.bashrc
+    source /home/vagrant/.bashrc
+
+    echo "----------------------------------------"
     echo "."
+    echo "ECLIPSE CHE: DOWNLOADING DOCKER REGISTRY"
+    echo "             50MB: SILENT OUTPUT        "
     echo "."
-    echo "ECLIPSE CHE: PREPPING SERVER"
+    echo "----------------------------------------"
+    docker pull registry:2 &>/dev/null
+
+    echo "---------------------------------"
     echo "."
+    echo "ECLIPSE CHE: PREPPING SERVER ~10s"
     echo "."
+    echo "---------------------------------"
 
     if [ -n "$HTTP_PROXY" ]; then
         sed -i "s|http.proxy=|http.proxy=${HTTP_PROXY}|" /home/vagrant/eclipse-che-*/conf/che.properties
     fi
     if [ -n "$HTTPS_PROXY" ]; then
         sed -i "s|https.proxy=|https.proxy=${HTTPS_PROXY}|"  /home/vagrant/eclipse-che-*/conf/che.properties
     fi
-    echo vagrant | sudo -S -E -u vagrant /home/vagrant/eclipse-che-*/bin/che.sh --remote:192.168.28.30 --skip:client -g start
+    echo vagrant | sudo -S -E -u vagrant /home/vagrant/eclipse-che-*/bin/che.sh --remote:${IP} --skip:client -g start &>/dev/null
   SHELL
 
   config.vm.provision "shell" do |s|
   	s.inline = $script
-  	s.args = [$http_proxy, $https_proxy]
+  	s.args = [$http_proxy, $https_proxy, $che_version, $ip]
   end
 
+  $script2 = <<-SHELL
+    IP=$1
+    counter=0
+    while [ true ]; do
+      curl -v http://${IP}:8080/dashboard &>/dev/null
+      exitcode=$?
+
+      if [ $exitcode == "0" ]; then
+        echo "----------------------------------------"
+        echo "."
+        echo "ECLIPSE CHE: SERVER BOOTED AND REACHABLE"
+        echo "AVAILABLE:   http://$IP:8080  "
+        echo "."
+        echo "----------------------------------------"
+        exit 0
+      fi 
+
+      # If we are not awake after 30 seconds, restart server
+      if [ $counter == "5" ]; then
+        echo "-----------------------------------------------"
+        echo "."
+        echo "ECLIPSE IDE: SERVER NOT RESPONSIVE -- REBOOTING"
+        echo "."
+        echo "-----------------------------------------------"
+        export JAVA_HOME=/usr &>/dev/null
+        echo vagrant | sudo -S -E -u vagrant /home/vagrant/eclipse-che-*/bin/che.sh --remote:$IP --skip:client -g start &>/dev/null
+      fi
+
+      let counter=counter+1
+      sleep 5
+    done
```

Comment:
what if server will not be able to boot at all? there should be exit after some amount of tries.
Also on some environments server may start longer than 30 secs and in that case you will be in a race with wait starting & reboot server. I think we have to let server few mins to start and than check is is running, try restart and if restart didn;t help we have to exit that loop.